crates/web-plugins/didcomm-messaging/keystore/src/lib.rs

```diff
@@ -34,6 +18,81 @@ pub struct KeyStore<'a> {
     keys: Vec<Jwk>,
 }
 
+use chacha20poly1305::{
+    aead::{generic_array::GenericArray, Aead, AeadCore, OsRng},
+    ChaCha20Poly1305, KeyInit,
+};
+
+use log::{debug, info}; 
+use secrecy::{ExposeSecret, SecretString};
+use zeroize::Zeroize;
+
+struct FileSystemKeystore {
+    key: SecretString, // Store key securely using secrecy crate
+    nonce: Vec<u8>,
+}
+
+impl FileSystemKeystore {
+    fn encrypt(mut self, secret: KeyStore) -> Result<(), KeystoreError> {
+        let key = self.key.expose_secret(); // Access key securely
```

Comment:
@Blindspot22 These are the changes i am talking about, this function is actually not used anywhere in the code and there is no test for it still